src/ariadne_auth/authz.py

```diff
@@ -132,7 +137,11 @@ def resolve(
         info: GraphQLResolveInfo,
         **kwargs: Any,
     ) -> Any:
-        if not isinstance(next_, self.PermissionChecker) and self.global_permissions:
+        try:
+            resolver = info.parent_type.fields[info.field_name].resolve
+        except (AttributeError, KeyError):
+            resolver = next_
```

Comment:
`AttributeError` was added to keep tests passing and keep the PR concise for review. Happy to update if the PR looks good.
`KeyError` was out of caution as I am unsure if this is the right way to look the resolver up. Suggestions welcome!

Reply:
Thanks @reallistic.
The PR looks good to me.

Honestly I don't have strong opinion about the right way to look the resolver up.
There is always a possibility to create a PR when something smart will comes to our minds :)

Reply:
Nice we can keep it as is then and fix/update things as needed.